docs/permissions.rst

```diff
@@ -235,26 +238,289 @@ to execute `Magpie` application. Effectively, when the active session correspond
+        :attr:`Access.DENY`, the resolved access is marked as denied. If every equally prioritized :term:`Group`
+        indicate :attr:`Access.ALLOW`, then access is granted to the :term:`User`.
+
+    2.3 Otherwise, the highest priority :term:`Group` dictates the :class:`Access` resolution. This can
+        *potentially revert* a previous :term:`Group` decision.
+
+
+The specific use case of step (2.3) is intended to give higher resolution precedence to *generic* :term:`Groups` over
+the *special* ``MAGPIE_ANONYMOUS_GROUP`` definition. This means that a *generic* :term:`Group` with a
+:term:`Permission` affected by :attr:`Access.ALLOW` modifier can override *special* ``MAGPIE_ANONYMOUS_GROUP`` that
+would have :attr:`Access.DENY` for the same :term:`Resource`, although resolution is opposite in every other
+situation. The reason for this exception is due to the nature of ``MAGPIE_ANONYMOUS_GROUP`` membership that is being
+automatically applied to every :term:`User` in order to also grant them `Public Access`_ to any :term:`Resource`
+marked as accessible to anyone. Since that *special* :term:`Group` also represents *"unauthenticated users"*, it is
+both counter intuitive and not practical to equally resolve conflicting :term:`Inherited Permissions` as it is
+naturally expected that an authenticated :term:`User` with specific :term:`Group` memberships should receive higher
+access privileges than its unauthenticated counterpart in case of contradictory :term:`Permissions` across both
+:term:`Group`. In other words, when a :term:`Resource` is blocked to the open public, it is expected that a
+:term:`User` that would obtain access to that :term:`Resource` through another one of its :term:`Group` memberships
+doesn't remain denied access due to its implicit ``MAGPIE_ANONYMOUS_GROUP`` membership. Step (2.3) handles this edge
+case specifically.
+
+Every *generic* :term:`Group` share the same priority, and will therefore resolve conflicting :class:`Access` using the
+normal step conditions and prioritizing :attr:`Access.DENY`.
+
+When resolving only :term:`Inherited Permissions`, the procedure stops here and provides the applicable result if any
+was found, with the corresponding ``reason``. An empty set of :term:`Permission` is returned if none could be found.
+
+When instead resolving :term:`Effective Permissions`, the above process continues by rewinding the parent
+:term:`Resource` hierarchy until the first :term:`Permission` is found. Only on the first iteration (when the targeted
+:term:`Resource` is the same as the one looked for potential :term:`Inherited Permissions`) does :attr:`Scope.MATCH`
+take effect. Only :attr:`Scope.RECURSIVE` are considered afterwards. When a :term:`Permission` is found, the process
+immediately completes if :attr:`Access.DENY` results from the previous resolution steps. Otherwise, the process still
```

Comment:
If the found permission is for User, yes allow/deny is returned directly because that the highest possible priority, you are right.
This actually made me think of one still case not handled... 

If we find a Group Allow/Deny, we must continue rewinding regardless in case there is a User Deny/Allow-Recursive higher up that reverts it. 
I could confirm that this doesn't work currently.

Reply:
Good point. But given that the current code doesn't account for user permissions upward, I'm wondering why do you rewind right now? Am I missing something else?

Reply:
doc fixed with latest resolution method